### DISCUSSION TITLE
is there an equivalent to the PHI function of nonmem ?

### DISCUSSION THREAD

**Discussion by CombetR** [metrumresearchgroup/mrgsolve#1153] [Q&A] [answered]
Hello , 

I'm currently translating a nonmem model to R using mrgsolve.
This NonMem model uses the PHI function , which gives the value of the cumulative distribution function, directly in the nonmem outputs.
Is there an equivalent in mrgsolve that can be integrated directly into .cpp model?

Thanks  a lot !

**Comment by kylebaron** [comment, marked as answer]
Hi @CombetR  - 

Yes, there is. 

If you use the `Rcpp` plugin, you get access to all the `d/p/q/r` functions that you would call in R.  Usually we want this to call things like `rnorm()`, but you can call `pnorm()` too:

``` r
pnorm(c(-1.96, 1.96), 0, 1)
#> [1] 0.0249979 0.9750021
```

<sup>Created on 2024-01-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Here's how to use the plugin:  https://mrgsolve.org/user-guide/plugins.html#sec-plugin-rcpp

 Here's how you can call this in your model: 

``` r
library(mrgsolve)
#> 
#> Attaching package: 'mrgsolve'
#> The following object is masked from 'package:stats':
#> 
#>     filter

code <- '
$PLUGIN Rcpp

$ERROR
capture lower = R::pnorm(-1.96, 0.0, 1.0, 1, 0); 
capture upper = R::pnorm( 1.96, 0.0, 1.0, 1, 0); 
'

mod <- mcode("pnorm", code)
#> Building pnorm ...
#> done.

mrgsim(mod, end = 0)
#> Model:  pnorm 
#> Dim:    1 x 4 
#> Time:   0 to 0 
#> ID:     1 
#>     ID time lower upper
#> 1:   1    0 0.025 0.975
```

<sup>Created on 2024-01-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

The signature for `R::pnorm()` is `double, double, double, int, int` but you can use `bool` for the last two arguments, too. You just need to pass something for every argument you see in `stats::pnorm()`

``` r
args(pnorm)
#> function (q, mean = 0, sd = 1, lower.tail = TRUE, log.p = FALSE) 
#> NULL
```

<sup>Created on 2024-01-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I think there is a vectorized version of this too; just showing you the version for doing this for a single value.

**Reply by CombetR** [reply]
Thanks a lot Kyle !
